Description: Blocks default behavior of touchstart events, which on iOS can
trigger gestures that refresh the page or navigate back and forth.

--- a/src/plugins/platforms/wasm/qwasmwindow.cpp
+++ b/src/plugins/platforms/wasm/qwasmwindow.cpp
@@ -132,6 +132,14 @@
     m_keyDownCallback =
             std::make_unique<qstdweb::EventCallback>(m_qtWindow, "keydown", keyCallback);
     m_keyUpCallback = std::make_unique<qstdweb::EventCallback>(m_qtWindow, "keyup", keyCallback);
+
+    // Drawpile patch: prevent gestures on iOS from going back a page. This also
+    // prevents scrolling the page, but we don't want that either, so it's fine.
+    m_touchStartCallback = std::make_unique<qstdweb::EventCallback>(
+            m_qtWindow, "touchstart", [](emscripten::val event) {
+                event.call<void>("preventDefault");
+                return false;
+            });
 }

 QWasmWindow::~QWasmWindow()
--- a/src/plugins/platforms/wasm/qwasmwindow.h
+++ b/src/plugins/platforms/wasm/qwasmwindow.h
@@ -142,6 +142,8 @@

     std::unique_ptr<qstdweb::EventCallback> m_wheelEventCallback;

+    std::unique_ptr<qstdweb::EventCallback> m_touchStartCallback;
+
     Qt::WindowStates m_state = Qt::WindowNoState;
     Qt::WindowStates m_previousWindowState = Qt::WindowNoState;
